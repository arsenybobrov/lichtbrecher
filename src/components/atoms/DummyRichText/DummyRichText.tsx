import React from 'react';
import uniqueId from 'lodash/uniqueId';

const RichTextElementMap: Record<string, string> = {
  paragraph: 'p',
};

export interface RichTextElementType {
  type: string;
  text: string;
  key?: string;
}

interface RichTextProps {
  content: Array<RichTextElementType>;
}

const renderRichTextElement = ({ type, text, key }: RichTextElementType) => {
  if (type in RichTextElementMap) {
    return React.createElement(RichTextElementMap[type], { key }, text);
  }
  return <pre key={key}>{JSON.stringify({ type, text })}</pre>;
};

const DummyRichText: React.FC<RichTextProps> = ({ content }) => (
  <>
    {content.map((elem) => (
      renderRichTextElement({ ...elem, key: uniqueId(elem.type) })
    ))}
  </>
);

export default DummyRichText;
