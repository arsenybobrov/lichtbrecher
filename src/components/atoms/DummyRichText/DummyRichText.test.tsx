import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import DummyRichText from './DummyRichText';

// eslint-disable-next-line import/prefer-default-export
export const paragraphsMock = [
  {
    type: 'paragraph',
    text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  },
  {
    type: 'paragraph',
    text: 'Aliquam dui turpis, ultricies quis semper vel, facilisis in leo.',
  },
  {
    type: 'paragraph',
    text: 'In aliquet orci et egestas sagittis.',
  },
  {
    type: 'paragraph',
    text:
      'Phasellus pretium, elit ac pellentesque porta, lacus sapien laoreet neque, nec bibendum odio leo sollicitudin tellus.',
  },
];

const listMock = [
  {
    type: 'list-item',
    text: 'first',
  },
  {
    type: 'list-item',
    text: 'second',
  },
  {
    type: 'list-item',
    text: 'third',
  },
];

describe('nextLink snapshot', () => {
  it.each`
    name                              | mock
    ${'RichText with paragraphs'}     | ${paragraphsMock}
    ${'Richtext with unordered list'} | ${listMock}
  `('$name', ({ mock }) => {
  const tree = renderer.create(<DummyRichText content={mock} />).toJSON();
  expect(tree).toMatchSnapshot();
});
});
