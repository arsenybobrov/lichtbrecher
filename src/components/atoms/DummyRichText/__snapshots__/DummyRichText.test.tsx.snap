// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nextLink snapshot RichText with paragraphs 1`] = `
Array [
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </p>,
  <p>
    Aliquam dui turpis, ultricies quis semper vel, facilisis in leo.
  </p>,
  <p>
    In aliquet orci et egestas sagittis.
  </p>,
  <p>
    Phasellus pretium, elit ac pellentesque porta, lacus sapien laoreet neque, nec bibendum odio leo sollicitudin tellus.
  </p>,
]
`;

exports[`nextLink snapshot Richtext with unordered list 1`] = `
Array [
  <pre>
    {"type":"list-item","text":"first"}
  </pre>,
  <pre>
    {"type":"list-item","text":"second"}
  </pre>,
  <pre>
    {"type":"list-item","text":"third"}
  </pre>,
]
`;
